### DISCUSSION TITLE
Hybrid Mode: Can I run the admin separately from the control plane and data plane services?

### DISCUSSION THREAD

**Discussion by chrisforrette** [Kong/kong#8053] [Help]
I've been running Kong Gateway in Hybrid Mode in AWS ECS using AWS load balancers and CodeDeploy. I have 3 separate ECS services: one for control plane, one for data plane, and one for the admin API. The reason for doing this is that running from multiple ports from the same container in ECS is a bit limited. Anyway, I've found this composition to be problematic in that updated configuration on the isolated admin service doesn't seem to reliably make it down to the data plane service (e.g. adding a route to the admin, accessing the route from the data plane 404s).

I've also noticed that the control plane service won't even start up without the admin service running.

Thinking about it though, any number of control plane instances could be running in a cluster, and any one of them could be accessed to update config, so theoretically there is some mechanism for them all to be made aware of it. I'm wondering if I might just be missing a bit of configuration.

Is there a way to run the admin service in isolation like this?

**Comment by dndx** [comment]
Hello @chrisforrette,

Which version of Kong are you using? We fixed the issue of event propagation when running CP and Admin API on different hosts a while ago. Could you try with the latest Kong release and see if the problem still exists?

**Reply by chrisforrette** [reply]
@dndx Hey there! We are running version 2.5.1. I tried upgrading to the latest (2.6.0) and observed the same behavior.

However, I was able to get it working by setting `KONG_ROLE` to `control_plane` on the admin service.
